src/common/io/read.c

```diff
@@ -196,6 +199,16 @@ ioRead(IoRead *this, Buffer *buffer)
     FUNCTION_LOG_RETURN(SIZE, outputRemains - bufRemains(buffer));
 }
 
+/***********************************************************************************************************************************
+Same as ioRead(), but returns the actual read size.
+***********************************************************************************************************************************/
+FN_EXTERN size_t
+ioReadSize(IoRead *this, Buffer *buffer)
```

Comment:
Almost all pgbackrest functions log their arguments and return value. I suggest to add logging in ioReadSize too

Reply:
It is done by ioRead